packages/@aws-cdk/assertions/lib/private/resources.ts

```diff
@@ -32,11 +32,12 @@ export function hasResource(template: Template, type: string, props: any): strin
 }
 
 export function hasResourceProperties(template: Template, type: string, props: any): string | void {
-  // amended needs to be a deep copy to avoid modifying the template.
-  let amended = JSON.parse(JSON.stringify(template));
+  let amended = template;
 
   // special case to exclude AbsentMatch because adding an empty Properties object will affect its evaluation.
   if (!Matcher.isMatcher(props) || !(props instanceof AbsentMatch)) {
+    // amended needs to be a deep copy to avoid modifying the template.
+    amended = JSON.parse(JSON.stringify(template));
```

Comment:
This is a bonus little optimization that I noticed while in the code. Unconditionally stringifying and parsing a template will negatively impact performance; those are not cheap operations.